app/decorators/service_provider_session_decorator.rb

```diff
@@ -34,6 +34,22 @@ def sp_logo
   end
 
   def sp_logo_url
+    if FeatureManagement.logo_upload_enabled? && sp.logo_key.present?
+      s3_logo_url(sp)
+    else
+      legacy_logo_url
+    end
+  end
+
+  def s3_logo_url(service_provider)
+    region = Figaro.env.aws_region
+    bucket = Figaro.env.aws_logo_bucket
+    key = service_provider.logo_key
+
+    "https://s3.#{region}.amazonaws.com/#{bucket}/#{key}"
```

Comment:
Are we going to have a Cloudfront distribution for these?

Reply:
Not at this time. 😢

Reply:
We should uh, get on that because having a lot of web traffic directly to an S3 bucket can get expensive quickly (obviously for a follow-up PR)

Reply:
I absolutely agree.